### PR TITLE
feat(KONFLUX-2157): prepare the App to deal with the jobs failing during cluster creation

### PR DESCRIPTION
**Pull request by dheerajodha**
* Modifies the comment message when the prow job fails during
   cluster creation by attaching the content of `build-log.txt` file.
* Remove the klog and replace with the one that we've configured
* Slightly refactored `updateCommentWithFailedTestCasesReport()`
  function, by moving log messages to correct places so they're
  not misleading in case no updates are made.
* Attached prow job's URL to logger for easier debugging
* Updated the go.mod and go.sum files.

`Signed-off-by: Dheeraj<djodha@redhat.com>`

### CODE REVIEW

**Comment by dheerajodha** [comment]
Related PR on the qe-tools repo: https://github.com/redhat-appstudio/qe-tools/pull/85, that adds the functionality to detect issues from outside of the e2e workflow.

**Comment by dheerajodha** [comment]
_(This is what the comment looks like)_

:rotating_light: **This is a CI system failure, please consult with the QE team.**

 <details><summary>Click to view logs</summary><br><pre>[36mINFO[0m[2024-02-11T12:16:41Z] ci-operator version v20240210-10eb2346f      
[36mINFO[0m[2024-02-11T12:16:41Z] Loading configuration from https://config.ci.openshift.org for redhat-appstudio/integration-service@main 
[36mINFO[0m[2024-02-11T12:16:42Z] Resolved source https://github.com/redhat-appstudio/integration-service to main@828b9406, merging: #562 2f5b49b3 @yftacherzog 
[33mWARN[0m[2024-02-11T12:16:42Z] skipped directory "..2024_02_11_12_16_34.1237755862" when creating secret from directory "/secrets/ci-pull-credentials" 
[36mINFO[0m[2024-02-11T12:16:42Z] Requesting 4.13 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=amd64&channel=fast-4.13 
[36mINFO[0m[2024-02-11T12:16:42Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release@sha256:c39085b26665ed9877e208c123ff147cf7939dde4eca79a4f00fda8ea5b13dc9 
[36mINFO[0m[2024-02-11T12:16:42Z] Using namespace https://console.build02.ci.openshift.org/k8s/cluster/projects/ci-op-097gnk16 
[36mINFO[0m[2024-02-11T12:16:42Z] Running [input:root], [input:e2e-test-runner], [release:latest-integration-service-e2e], src, redhat-appstudio-integration-service-image, integration-service-e2e 
[36mINFO[0m[2024-02-11T12:16:42Z] Tagging openshift/release:golang-1.19 into pipeline:root. 
[36mINFO[0m[2024-02-11T12:16:42Z] Tagging ci/redhat-appstudio:e2e-test-runner into pipeline:e2e-test-runner. 
[36mINFO[0m[2024-02-11T12:16:42Z] Building src                                 
[36mINFO[0m[2024-02-11T12:16:42Z] Found existing build "src-amd64"             
[36mINFO[0m[2024-02-11T12:16:42Z] build didn't start running within 30m0s (phase: Pending):
Found 14 events for Pod src-amd64-build:
* 2024-02-11T12:08:46Z 2x default-scheduler: 0/19 nodes are available: 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/19 nodes are available: 19 Preemption is not helpful for scheduling.
* 2024-02-11T11:47:45Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}
* 2024-02-11T11:48:16Z 1x default-scheduler: 0/20 nodes are available: 1 node(s) had untolerated taint {node.kubernetes.io/not-ready: }, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T11:52:13Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 not ready for scale-up
* 2024-02-11T12:13:46Z 3x default-scheduler: 0/20 nodes are available: 1 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T11:53:21Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }
* 2024-02-11T11:54:27Z 1x cluster-autoscaler: pod didn't trigger scale-up: 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }
* 2024-02-11T11:55:39Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up
* 2024-02-11T11:56:48Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up
* 2024-02-11T12:03:17Z 1x default-scheduler: 0/20 nodes are available: 1 node(s) were unschedulable, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T12:05:00Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }
* 2024-02-11T12:09:39Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }
* 2024-02-11T12:11:54Z 1x cluster-autoscaler: pod didn't trigger scale-up: 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}
* 2024-02-11T12:16:24Z 2x cluster-autoscaler: (combined from similar events): pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker} 
[36mINFO[0m[2024-02-11T12:16:42Z] Importing release image latest-integration-service-e2e. 
[36mINFO[0m[2024-02-11T12:16:42Z] Resolved release latest-integration-service-e2e to quay.io/openshift-release-dev/ocp-release:4.13.32-x86_64 
[36mINFO[0m[2024-02-11T12:17:13Z] Imported release 4.13.32 created at 2024-02-02 09:15:41 +0000 UTC with 183 images to tag release:latest-integration-service-e2e 
[36mINFO[0m[2024-02-11T12:17:13Z] Ran for 31s                                  
[31mERRO[0m[2024-02-11T12:17:13Z] Some steps failed:                           
[31mERRO[0m[2024-02-11T12:17:13Z] 
  * could not run steps: step src failed: error occurred handling build src-amd64: build didn't start running within 30m0s (phase: Pending):
Found 14 events for Pod src-amd64-build:
* 2024-02-11T12:08:46Z 2x default-scheduler: 0/19 nodes are available: 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/19 nodes are available: 19 Preemption is not helpful for scheduling.
* 2024-02-11T11:47:45Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}
* 2024-02-11T11:48:16Z 1x default-scheduler: 0/20 nodes are available: 1 node(s) had untolerated taint {node.kubernetes.io/not-ready: }, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T11:52:13Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 not ready for scale-up
* 2024-02-11T12:13:46Z 3x default-scheduler: 0/20 nodes are available: 1 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T11:53:21Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }
* 2024-02-11T11:54:27Z 1x cluster-autoscaler: pod didn't trigger scale-up: 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }
* 2024-02-11T11:55:39Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up
* 2024-02-11T11:56:48Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up
* 2024-02-11T12:03:17Z 1x default-scheduler: 0/20 nodes are available: 1 node(s) were unschedulable, 2 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 4 node(s) didn't match Pod's node affinity/selector, 7 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }. no new claims to deallocate, preemption: 0/20 nodes are available: 20 Preemption is not helpful for scheduling.
* 2024-02-11T12:05:00Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) didn't match Pod's node affinity/selector, 3 not ready for scale-up, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }
* 2024-02-11T12:09:39Z 1x cluster-autoscaler: pod didn't trigger scale-up: 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }
* 2024-02-11T12:11:54Z 1x cluster-autoscaler: pod didn't trigger scale-up: 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker}
* 2024-02-11T12:16:24Z 2x cluster-autoscaler: (combined from similar events): pod didn't trigger scale-up: 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-longtests-worker: ci-longtests-worker}, 3 not ready for scale-up, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-tests-worker: }, 1 node(s) had untolerated taint {node-role.kubernetes.io/infra: }, 3 node(s) didn't match Pod's node affinity/selector, 3 node(s) had untolerated taint {node-role.kubernetes.io/ci-prowjobs-worker: ci-prowjobs-worker} 
[36mINFO[0m[2024-02-11T12:17:13Z] Reporting job state 'failed' with reason 'executing_graph:step_failed:cloning_source:pod_pending' 
</pre></details>

-------------------------------

The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
ci/prow/redhat-appstudio-e2e | 4336ccc68894ef7f2695a8fafb7f602a9840c1f3 | [link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_integration-service/562/pull-ci-redhat-appstudio-integration-service-main-integration-service-e2e/1756653451292446720) | true | `/test redhat-appstudio-e2e`
ci/prow/images | 4336ccc68894ef7f2695a8fafb7f602a9840c1f3 | [link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_e2e-tests/1019/pull-ci-redhat-appstudio-e2e-tests-main-images/1751926606277906432) | true | `/test redhat-appstudio-e2e`

[Full PR test history](https://prow.ci.openshift.org/pr-history?org=redhat-appstudio&repo=e2e-tests&pr=1010). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:stuartwdouglas).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: issue_comment.go [hunk 1]
```diff
@@ -30,15 +30,15 @@ import (
 	"github.com/redhat-appstudio/qe-tools/pkg/prow"
 	"github.com/rs/zerolog"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/klog/v2"
 )
 
 const (
-	targetAuthor             = "openshift-ci[bot]"
+	targetAuthor             = "dheerajodha"
```
Comment (psturc) [inline issue_comment.go]: should we revert this?

File: issue_comment.go [hunk 2]
```diff
@@ -191,8 +191,22 @@ func setHeaderString(logger zerolog.Logger, overallJUnitSuites *reporters.JUnitT
 // 'failedTestCaseNames' field with the names of failed test cases
 // within the given JUnitTestSuites. It does nothing, if the given
 // JUnitTestSuites is nil.
```
Comment (psturc) [inline issue_comment.go]: nit: should the comment be updated?

**Review by psturc — CHANGES_REQUESTED** [review]
just 2 things, otherwise looks good!

File: issue_comment.go (hunk 2 above)
Comment (dheerajodha) [reply issue_comment.go]: thanks for catching this! fixed.

File: issue_comment.go (hunk 1 above)
Comment (dheerajodha) [reply issue_comment.go]: oops, yes sir. fixed

**Review by psturc — APPROVED** [review]

**Pull request merged**